### PR TITLE
Introduce regional site

### PR DESCRIPTION
**Pull request by njam**
- Denkmal_Site_Abstract
- Denkmal_Site_Default
- Denkmal_Site_Region_Abstract
- Denkmal_Site_Region_Basel
- Denkmal_Site_Region_Graz
- Admin_Site extends Denkmal_Site_Abstract

### CODE REVIEW

**Comment by njam** [comment]
@fvovan please review

File: library/Denkmal/library/Denkmal/Site/Default.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * @return Denkmal_Model_Region
+     * @throws CM_Exception
+     */
+    public function getRegion() {
+        throw new CM_Exception('Site has no region');
+    }
+
```
Comment (fvovan) [inline library/Denkmal/library/Denkmal/Site/Default.php]: Do we need these two methods?

File: library/Denkmal/library/Denkmal/Site/Region/Basel.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+class Denkmal_Site_Region_Basel extends Denkmal_Site_Region_Abstract {
+
+    public function _getRegionSlug() {
```
Comment (fvovan) [inline library/Denkmal/library/Denkmal/Site/Region/Basel.php]: I would keep visibility modifier the same as for parent's method

File: library/Denkmal/library/Denkmal/Site/Default.php (hunk 1 above)
Comment (njam) [inline library/Denkmal/library/Denkmal/Site/Default.php]: The idea is to have a consistent interface (`hasRegion` and `getRegion`) on all site instances. This way we can call the two methods across all of denkmal's templates etc.

File: library/Denkmal/library/Denkmal/Site/Region/Abstract.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    public function getRegion() {
+        $slug = $this->_getRegionSlug();
+        $region = Denkmal_Model_Region::findBySlug($slug);
+        if (null === $region) {
+            throw new CM_Exception('Cannot find region with slug `' . $slug . '`');
+        }
+        return $region;
+    }
+
+    abstract protected function _getRegionSlug();
```
Comment (fvovan) [inline library/Denkmal/library/Denkmal/Site/Region/Abstract.php]: please add docu

**Comment by fvovan** [comment]
generally looks good.
the only note is that if we splitted config into 2 parts (config itself and `Setting` class). Not sure if  its methods could exist in `Default_Site`.

**Pull request merged**